packages/controller/src/dbClient.ts

```diff
@@ -30,14 +30,15 @@ if (endpoint && !adminSecret) {
 const client = endpoint
   ? new GraphQLClient(endpoint, {
       headers: {
-        "x-hasura-admin-secret": adminSecret!
+        "x-hasura-admin-secret": adminSecret
       }
     })
   : null;
 
 export type PromiseReturnType<T> = T extends PromiseLike<infer U> ? U : T;
-export type ClientResponse<T extends (args?: any) => {}> = PromiseReturnType<
-  ReturnType<T>
->;
+export type ClientResponse<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  T extends (args?: any) => Record<string, any>
+> = PromiseReturnType<ReturnType<T>>;
```

Comment:
I am curious about what this is doing. Naively I would have thought that the above is equivalent to:
```suggestion
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type ClientResponse = PromiseReturnType<Record<string, any>>;
```